docs/execution.rst

```diff
@@ -323,24 +323,14 @@ threading strategy you use is up to your data fetcher code.
 The following code uses the standard Java ``java.util.concurrent.ForkJoinPool.commonPool()`` thread executor to supply values in another
 thread.
 
-.. code-block:: java
-
-        DataFetcher userDataFetcher = new DataFetcher() {
-            @Override
-            public Object get(DataFetchingEnvironment environment) {
-                CompletableFuture<User> userPromise = CompletableFuture.supplyAsync(() -> {
-                    return fetchUserViaHttp(environment.getArgument("userId"));
-                });
-                return userPromise;
-            }
-        };
+You can use ``graphql.schema.AsynchronousDataFetcher.async(DataFetcher<T>)`` to wrap a ``DataFetcher`` so that it runs asynchronously.
```

Comment:
Please reinstate the original documentation.  We want to show it more from first principles and how one might write their own async data fetcher

Feel free to add a futher section that might say something like

    There is a helpful shortcur in graphql-java to create asynchronous data fetchers. 
    Use ``graphql.schema.AsynchronousDataFetcher.async(DataFetcher<T>)`` to wrap a 
    ``DataFetcher``  as such .....